SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskLogManager.java

```diff
@@ -93,15 +94,42 @@ private boolean writeS3MetadataFileForRotatedFiles(boolean finished) {
   }
 
   private void writeLogrotateFile() {
-    log.info("Writing logrotate configuration file to {}", getLogrotateConfPath());
+    log.info("Writing non-hourly logrotate configuration file to {}", getLogrotateConfPath());
     templateManager.writeLogrotateFile(getLogrotateConfPath(), new LogrotateTemplateContext(configuration, taskDefinition));
 
-    if (logrotateFrequency.getCronSchedule().isPresent()) {
-      log.info("Writing logrotate cron entry with schedule '{}' to {}", logrotateFrequency.getCronSchedule().get(), getLogrotateCronPath());
-      templateManager.writeCronEntryForLogrotate(getLogrotateCronPath(), new LogrotateCronTemplateContext(configuration, taskDefinition, logrotateFrequency));
+    // Get the frequency and cron schedule for an additional file with an HOURLY schedule, if there is any
+    Optional<SingularityExecutorLogrotateFrequency> additionalFileFrequency = getAdditionalHourlyFile();
+
+    // if any additional file or the global setting has an hourly rotation, write a separate rotate config and force rotate using a cron schedule
+    if (additionalFileFrequency.isPresent() && additionalFileFrequency.get().getCronSchedule().isPresent() || logrotateFrequency.getCronSchedule().isPresent()) {
+      log.info("Writing hourly logrotate configuration file to {}", getLogrotateHourlyConfPath());
+      templateManager.writeHourlyLogrotateFile(getLogrotateHourlyConfPath(), new LogrotateTemplateContext(configuration, taskDefinition));
+
+      SingularityExecutorLogrotateFrequency freq = additionalFileFrequency.isPresent() && additionalFileFrequency.get().getCronSchedule().isPresent() ?
+          additionalFileFrequency.get() : logrotateFrequency;
+
+      String cronScheduleString = freq.getCronSchedule().isPresent() ? freq.getCronSchedule().get() : "Error in cron schedule"; // This should never evaluate to false
+
+      log.info("Writing logrotate cron entry with schedule '{}' to {}", cronScheduleString, getLogrotateCronPath());
+      templateManager.writeCronEntryForLogrotate(getLogrotateCronPath(), new LogrotateCronTemplateContext(configuration, taskDefinition, freq));
     }
   }
 
+  private Optional<SingularityExecutorLogrotateFrequency> getAdditionalHourlyFile() {
+    // Get the cron schedule for an additional file with an HOURLY schedule, if there is any
+    java.util.Optional<SingularityExecutorLogrotateAdditionalFile> additionalHourlyFile = configuration.getLogrotateAdditionalFiles()
```

Comment:
nit, while util.Optional is the newer/nicer. It's probably easier for us to stick with guava optional until we do a consolidated swap across the whole repo.